### PR TITLE
Docs/403 add hackathon work to the repo

### PR DESCRIPTION
**Pull request by Carsopre**
## Issue addressed
Solves #403 

## What has been done?
- Moved hackathon directory into examples.
- Created summary jupyter notebook with runnable snippets.
- Added cleaned up functions whilst building documentation via sphinx.

## Checklist
- [x] Tests are either added or updated.
- [x] Branch is up to date with `master`.
- [x] Updated documentation if needed.

## Additional Notes (optional)
User Question 2 is not really complete, however this work should be extended by our domain experts.

### CODE REVIEW

File: examples/hackathons/20240326_hackathon/README.md
Comment (ArdtK) [inline examples/hackathons/20240326_hackathon/README.md:1]: This readme has quite some overlap with summary notebook `summary_2024q1.ipynb`. Is it worth having both?

File: examples/hackathons/20240326_hackathon/user_question_1/README.md
Comment (ArdtK) [inline examples/hackathons/20240326_hackathon/user_question_1/README.md:1]: This documentation is quite extensive. I'm not sure if that's worth the effort for next hackathons. If we want to keep outputs of the hackathon, we could schedule it in a next sprint to formalize it.

**Review by ArdtK — APPROVED** [review]
Approved with some questions/comments

File: examples/hackathons/20240326_hackathon/README.md
Comment (Carsopre) [reply examples/hackathons/20240326_hackathon/README.md:1]: Yes, the jupyter notebook is unfortunately not everywhere supported whilst the `README.md` it's just a rich text file.

File: examples/hackathons/20240326_hackathon/user_question_1/README.md
Comment (Carsopre) [reply examples/hackathons/20240326_hackathon/user_question_1/README.md:1]: I partially agree. Whilst the `README.md` is not needed, we still require a document where each question is addressed individually on what needs to be resolved and how. In my opinion we can ditch these `README.md` with well-described jupyter notebooks.

**Pull request merged**